src/main/kotlin/build/buf/intellij/resolve/BufRootsProvider.kt

```diff
@@ -14,147 +14,37 @@
-                return null
-            }
-            val commitTempPath = Files.createTempDirectory(commitPath.parent, "tmp_" + mod.commit)
-            try {
-                for (path in manifest.getPaths()) {
-                    val canonicalPath = manifest.getCanonicalPath(path) ?: return null
-                    val modulePath = commitTempPath.resolve(path)
-                    if (!modulePath.parent.isDirectory()) {
-                        Files.createDirectories(modulePath.parent)
-                    }
-                    Files.copy(Path.of(canonicalPath), modulePath)
-                }
-                try {
-                    Files.move(commitTempPath, commitPath, StandardCopyOption.ATOMIC_MOVE, StandardCopyOption.REPLACE_EXISTING)
-                } catch (e: DirectoryNotEmptyException) {
-                    // This is expected if we raced with another process
-                } catch (e: FileSystemException) {
-                    // This can happen in a race - DirectoryNotEmpty
-                    val message = e.message ?: ""
-                    if (!message.contains("Directory not empty")) {
-                        LOG.warn("failed to move directory $commitTempPath to $commitPath: $e")
-                    }
-                }
-                return LocalFileSystem.getInstance().findFileByNioFile(commitPath)
-            } catch (e: Exception) {
-                LOG.warn("failed to copy buf v2 cache module $mod", e)
-                return null
-            } finally {
-                if (commitTempPath.exists()) {
-                    try {
-                        commitTempPath.delete(recursively = true)
-                    } catch (e: IOException) {
-                        LOG.warn("failed to delete temporary dir: $commitTempPath", e)
-                    }
-                }
-            }
-        }
-    }
-
-    /**
-     * Let's index all Buf modules
-     * */
-    override fun getRootsToWatch(project: Project): Collection<VirtualFile> {
-        val bufPath = getIDEAModCacheV2()
-        if (!bufPath.isDirectory() && !bufPath.toFile().mkdirs()) {
-            LOG.warn("failed to create directory path: ${bufPath.absolutePathString()}")
-        }
-        return listOfNotNull(
-            LocalFileSystem.getInstance().findFileByNioFile(bufPath),
-            getBufCacheFolderV1(),
-        )
-    }
+    override fun getRootsToWatch(project: Project): Collection<VirtualFile> = moduleCacheService.moduleDataRoots()
+        .mapNotNull { LocalFileSystem.getInstance().findFileByNioFile(it) }
 
     override fun getAdditionalProjectLibraries(project: Project): Collection<SyntheticLibrary> {
-        val bufPath = getIDEAModCacheV2()
-        if (!bufPath.isDirectory() && !bufPath.toFile().mkdirs()) {
-            LOG.warn("failed to create directory path: ${bufPath.absolutePathString()}")
+        if (DumbService.getInstance(project).isDumb) {
+            return emptyList()
         }
```

Comment:
I take it this is an optimization to skip some work when IDEA is in dumb mode?

Reply:
This is required to prevent exceptions being thrown while IDEA is rebuilding indexes.

Reply:
Oof. So this was just a bug in the previous version? (I asked about it because I noticed we didn't previously have any such preamble.)

Maybe a comment to that effect -- that the IDE is in "dumb mode" while building indexes, and that this is needed to prevent exceptions?

Reply:
The previous implementation didn't use indexes - it just added two libraries for the entire v1 or v2 module cache root directories. The new version uses data indexed from the `buf.lock` files to create libraries for each module dependency.

Will add a comment.